appservice/src/createAppService/SiteRuntimeStep.ts

```diff
@@ -86,12 +86,17 @@ export class SiteRuntimeStep extends AzureWizardPromptStep<IAppServiceWizardCont
         // this is to flatten the runtimes to one array
     }
 
-    private sortQuickPicksByRuntime(runtimeItems: IAzureQuickPickItem<ApplicationStack>[], recommendedRuntimes: string[]): IAzureQuickPickItem<ApplicationStack>[] {
-        function getPriority(item: IAzureQuickPickItem<ApplicationStack>): number {
-
-            const index: number = recommendedRuntimes.findIndex((runtime: string) => nonNullProp(item.data, 'name').includes(runtime));
-            return index === -1 ? recommendedRuntimes.length : index;
+    private sortQuickPicksByRuntime(runtimeQuickPicks: IAzureQuickPickItem<ApplicationStack>[], runtimeRecommendations: string[]): IAzureQuickPickItem<ApplicationStack>[] {
+        const recommendedQuickPicks: IAzureQuickPickItem<ApplicationStack>[] = [];
+        // go backwards so that when runtime is unshifted to front, the order is unaffected
+        for (let i: number = runtimeQuickPicks.length - 1; i >= 0; i = i - 1) {
+            for (const rt of runtimeRecommendations) {
+                if (nonNullProp(runtimeQuickPicks[i].data, 'name').toLocaleLowerCase().includes(rt.toLocaleLowerCase())) {
+                    recommendedQuickPicks.unshift(runtimeQuickPicks.splice(i, 1)[0]);
```

Comment:
I don't understand - is this sorting the list? It looks like you're moving `runtimeRecommendations` to the top of the list, but not actually sorting by version. Isn't that the same as what we had before your changes?

Reply:
Yep, it's only moving the recommended to the top.

And no, before that's what I _wanted_ it to do, but the bug came from the fact that the runtimes that were moved up to the top were getting swapped with whatever it replaced in order.  Which is why you'd sometimes see .NET core in the middle of the node runtimes or some of the other odd behaviors.

I thought we didn't really care about sorting, so I just left the default Portal ordering.